packages/gatsby/src/utils/__tests__/webpack.config.js

```diff
@@ -0,0 +1,115 @@
+jest.mock(`webpack`, () => {
+  return {
+    ...jest.requireActual(`webpack`),
+    DefinePlugin: jest.fn(),
+  }
+})
+jest.mock(`fs-extra`, () => {
+  const { readFileSync, ...rest } = jest.requireActual(`fs-extra`)
+  return {
+    ...rest,
+    readFileSync: jest
+      .fn()
+      .mockImplementation((...args) => readFileSync(...args)),
+  }
+})
+const { DefinePlugin } = require(`webpack`)
+const { readFileSync } = require(`fs-extra`)
+const webpackConfig = require(`../webpack.config`)
+const { store } = require(`../../redux`)
+
+beforeEach(() => {
+  DefinePlugin.mockClear()
+  readFileSync.mockClear()
+  store.dispatch({
+    type: `SET_PROGRAM`,
+    payload: {
+      extensions: [],
+      browserslist: [],
+      directory: process.cwd(),
+    },
+  })
+})
+
+const getConfig = (args = {}) =>
+  webpackConfig(
+    {
+      extensions: [`.js`],
+    },
+    process.cwd(),
+    `build-html`
+  )
+
+describe(`basic functionality`, () => {
+  it(`returns webpack config with expected shape`, async () => {
+    const config = await getConfig()
+
+    expect(config).toEqual(
+      expect.objectContaining(
+        [`context`, `entry`, `output`, `module`, `plugins`, `resolve`].reduce(
+          (merged, key) => {
+            merged[key] = expect.anything()
+            return merged
+          },
+          {}
+        )
+      )
+    )
+  })
+})
+
+describe(`environment variables`, () => {
+  afterEach(() => {
+    delete process.env.ACTIVE_ENV
```

Comment:
TIL that setting (and then unsetting via `process.env.SOME_ENV = undefined`) still makes that value available as a string, e.g.

```js
process.env.ACTIVE_ENV = `whatever`

// later

process.env.ACTIVE_ENV = undefined

expect(process.env.ACTIVE_ENV).toBe(`undefined`) // true
```